File: src/main/java/stdlib/LinkValue.java
package stdlib;

public class LinkValue implements Value {

    private final Object value;

    LinkValue(Object value) {
        this.value = value;
    }

    @Override
    public double asNumber() {
        return 0;
    }

    @Override
    public String asString() {
        return value.toString();
    }

    @Override
    public Object asLink() {
        return value;
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: src/main/java/stdlib/LinkValue.kt
package stdlib

class LinkValue internal constructor(private val value: Any) : Value {

    override fun asNumber(): Double {
        return 0.0
    }

    override fun asString(): String {
        return value.toString()
    }

    override fun asLink(): Any {
        return value
    }

    override fun toString(): String {
        return asString()
    }
}
